GoogleForCreators/web-stories-wp
# Editor: Improve behavior when right-clicking on multiple elements

## Context
Part of investigative work to see if we could better the right click menu ux with a better pattern for dismissing it when clicking outside.


## Summary
Adds a new approach to detect clicking outside of the context menu to dismiss it.


## Relevant Technical Choices
Used the `mousedown` capture event on the document to always detect the `mousedown` event even if components have disabled bubbling or prevented default. This seemed to cover all needed mouse interaction scenarios to dismiss the context menu.


## To-do
NA


## User-facing changes
You should now be able to right click between elements in the canvas without needing to have an intermediary click to dismiss the context menu. Other than that, should behave the same as before


## Testing Instructions






This PR can be tested by following these steps:

1. Open the editor and make sure you have several elements of different types
2. right click one element and see the relevant right click menu appear
3. right click an element of another type and see the correct right click menu appear
4. click anywhere off the right click menu and see that the right click menu gets dismissed (this should behave the same as before)


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #9426
